.circleci/config.yml

```diff
@@ -0,0 +1,46 @@
+version: 2.1
+jobs:
+  build:
+    docker:
+      - image: cimg/python:3.10.1
+    steps:
+      - checkout
+      - run:
+          name: Install
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r requirements.txt
+            pip install -e .
+      - save_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "requirements.txt" }}
+          paths:
+            - "venv"
+      - run:
+          name: Update submodules
+          command: |
+            . venv/bin/activate
+            git submodule update --init
+      - save_cache:
+          key: deps1-{{ .Branch }}-submodules
+          paths:
+            - "submodules"
+      - run:
+          name: Update repos.json
+          command: |
+            . venv/bin/activate
+            make repos.json
+      - save_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "repos.json" }}
+          paths:
+            - "repos.json"
```

Comment:
Why cache the `repos.json`?

Reply:
AFAIK, you don't need to cache any data between steps in a single workflow. Caching is only relevant for when you want to reuse data produced in a previous run in a new run. In fact, I imagine possible cache invalidation issues if `repos.json` is cached.

Reply:
Yeah, I got that wrong. My bad. Removed the caching for `repos.json`.